scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroupInternals.scala

```diff
@@ -115,6 +118,16 @@ private[dynamictls] object DynamicTLSPeerGroupInternals {
           val pipeline = ch.pipeline()
           val sslHandler = sslClientCtx.newHandler(ch.alloc())
 
+          socks5Config.foreach { config =>
+            val sock5Proxy = if (config.authConfig.isDefined) {
+              val authConfig = config.authConfig.get
+              new Socks5ProxyHandler(config.proxyAddress, authConfig.user, authConfig.password)
+            } else {
+              new Socks5ProxyHandler(config.proxyAddress)
+            }
```

Comment:
```suggestion
            val sock5Proxy = config.authConfig.fold {
               new Socks5ProxyHandler(config.proxyAddress)
            } { authConfig =>
              new Socks5ProxyHandler(config.proxyAddress, authConfig.user, authConfig.password)
            } 
```
Just to avoid the `.get`.